Binary/Loader/FileSystemLoader.php

```diff
@@ -34,15 +36,23 @@ class FileSystemLoader implements LoaderInterface
     protected $locator;
 
     /**
-     * @param MimeTypeGuesserInterface  $mimeGuesser
-     * @param ExtensionGuesserInterface $extensionGuesser
-     * @param LocatorInterface          $locator
+     * @param MimeTypeGuesserInterface|DeprecatedMimeTypeGuesserInterface $mimeGuesser
+     * @param MimeTypesInterface|DeprecatedExtensionGuesserInterface      $extensionGuesser
+     * @param LocatorInterface                                            $locator
      */
     public function __construct(
-        MimeTypeGuesserInterface $mimeGuesser,
-        ExtensionGuesserInterface $extensionGuesser,
+        $mimeGuesser,
+        $extensionGuesser,
         LocatorInterface $locator
     ) {
+        if (interface_exists(MimeTypeGuesserInterface::class) && $mimeGuesser instanceof DeprecatedMimeTypeGuesserInterface) {
```

Comment:
if we want to be strict, we need to throw an excception if $mimeGuesser / $extensionGuesser are not of one of the 2 classes. there is no type declaration anymore. something along `if (!$mimeGuesser instanceof MimeTypeGuesserInterface && !$mimeGuesser instanceof Deprecated...) {throw new InvalidArgumentException('...');}`